### DISCUSSION TITLE
SEO - Add a description meta tag to "article" page as well as category, tag and other special page (star, catalog, etc)

### DISCUSSION THREAD

**Discussion by JeremieLitzler** [vuepress-theme-hope/vuepress-theme-hope#2676] [Ideas]
Hi @Mister-Hope,

To improve the natural SEO, having the ability to set page description meta would be very nice.

I was thinking of something in the theme configuration where you give an object: 

```js
export default hopeTheme({`
  ...
  specialPages: {
    article: {
      metaDescription: "Some text..."
    },
    star: {
      metaDescription: "Some text..."
    },
    ...
  },
  categories: {
    "some-category": {
      metaDescription: "Some text..."
    }
  } 
  ...
});
``` 

or use a yaml file to set the meta tag as we can do it in the frontmatter?

```markdown
---
specialPages:
  article: 
    head:
      - [
          meta,
          {
            meta: "description",
            content: "Some description,
          },
        ]
...
category:
  - Gardening
    head:
      - [
          meta,
          {
            meta: "description",
            content: "Some description,
          },
        ]
tags:
  - January
    head:
      - [
          meta,
          {
            meta: "description",
            content: "Some description,
          },
        ]
---
```

What do you think is possible with the least effort on implementation and usage?

Thanks.

**Comment by Mister-Hope** [comment]
So what's the difference between the one you describe and setting description directly in frontmatter?

**Reply by JeremieLitzler** [reply]
If I'm correct, the pages 
- https://example.com/article/
- https://example.com/tag/
- https://example.com/tag/some-tag/
- https://example.com/category/some-category/
- https://example.com/start/

are special pages and I don't see how I can set a frontmatter for those.

I can definitely controld fully the markdown and the head tags from the markdown file for other page I create.

Am I missing something for the special pages?

**Reply by Mister-Hope** [reply]
I understand what you are talking, currently I am not expecting to add tons of options into the theme., so this is not supported.

Also, in the recent release, those pages are excluded from sitemap and feed.

I really do not think those pages are important and should be indexed by search engine. So what's the purpose of that?

**Reply by Mister-Hope** [reply]
For me, I would regard those as visitor-facing pages. Users visit them to get pages they want, while search engine omit thoese pages and directly get urls from sitemap. Also, those pages should not be included in feeds, neither they should be share links as  they are totally dynamic. So meta content is not important at all in my mind

Please explain why you want this and the backword reason driven you to do this.

**Reply by JeremieLitzler** [reply]
Hi @Mister-Hope 

Thanks for your answers. I understand your points.

Best regards.
